packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts

```diff
@@ -211,7 +260,40 @@ export class BedrockInvokeModel extends sfn.TaskStateBase {
         Output: this.props.output?.s3Location ? {
           S3Uri: `s3://${this.props.output.s3Location.bucketName}/${this.props.output.s3Location.objectKey}`,
         } : undefined,
+        GuardrailIdentifier: this.props.guardrail?.guardrailIdentifier,
+        GuardrailVersion: this.props.guardrail?.guardrailVersion,
+        Trace: this.props.traceEnabled === undefined
+          ? undefined
+          : this.props.traceEnabled
+            ? 'ENABLED'
+            : 'DISABLED',
       }),
     };
   }
+
+  private validateGuardrail(props: BedrockInvokeModelProps) {
+    if (!props.guardrail) return;
+
+    const { guardrailIdentifier, guardrailVersion } = props.guardrail;
+
+    if (!Token.isUnresolved(guardrailIdentifier)) {
+      const guardrailPattern = /^(([a-z0-9]+)|(arn:aws(-[^:]+)?:bedrock:[a-z0-9-]{1,20}:[0-9]{12}:guardrail\/[a-z0-9]+))$/;
+      if (!guardrailPattern.test(guardrailIdentifier)) {
+        throw new Error(`You must set guardrailIdentifier to the id or the arn of Guardrail, got ${guardrailIdentifier}`);
+      }
+      if (props.contentType !== 'application/json') {
+        throw new Error(`You must set contentType to \'application/json\' when using guardrail, got '${props.contentType}'.`);
+      }
+      if (guardrailIdentifier.length > 2048) {
+        throw new Error(`\`guardrailIdentifier\` length must be between 0 and 2048, got ${guardrailIdentifier.length}.`);
+      }
+
+    }
+
+    const guardrailVersionPattern = /^(([1-9][0-9]{0,7})|(DRAFT))$/;
```

Comment:
I think I would prefer a stricter check that just ensures the version is between 10 and 99999999 or is "DRAFT" instead of using regex here. Just to improve the readability of this code and make it easier to maintain in the future.

Reply:
Thank you. I have implemented it as follows.

https://github.com/aws/aws-cdk/pull/30426/files#diff-254b356f1a1b98a7034dec79302bc664e46a4de89a1fe14aa5c62d8982109dc2R15

Reply:
Awesome! The implementation looks good, I'd be happy to approve after the ARN class change is updated.